src/queryCache.js

```diff
@@ -14,6 +15,25 @@ import { defaultConfigRef } from './config'
 
 export const queryCache = makeQueryCache()
 
+export const defaultQueryCacheRef = {
+  current: queryCache
+}
+
+export const queryCacheContext = React.createContext(queryCache)
+
+export function useQueryCache() {
+  return React.useContext(queryCacheContext)
+}
+
+export function ReactQueryCacheProvider({ queryCache, children }) {
```

Comment:
What happens if `queryCache` changes here?

* If the old prop value was a `queryCache`, it's probably up to the parent to clean it up if necessary, it would be unsafe for this Provider to do so since it might still be in use
* If the old value was undefined, the old cache was created in this Provider, so this provider needs to clean it up

This relates to the more general issue of cleanup which I'll add a separate comment for.

Reply:
yup I've addressed this now (per my other comments)